harness/determined/common/experimental/model.py

```diff
@@ -114,7 +112,6 @@ def _hydrate(self, model_version: bindings.v1ModelVersion) -> None:
             model_version.checkpoint, self._session
         )
         self.metadata = model_version.metadata or {}
-        self.name = model_version.name or ""
```

Comment:
wait why was this removed?

Reply:
> no longer hydrates name during a ModelVersion.reload (to treat name like an id, not a mutable attribute)

This feels like the right call to me. If not, maybe it's subtle enough to huddle about?

Reply:
i see... but name _is_ mutable in the sense that we have a `set_name`... so it seems a little weird.